zebrad/src/components/health.rs

```diff
@@ -0,0 +1,115 @@
+//! A simple HTTP health endpoint for Zebra.
+
+use std::net::SocketAddr;
+use std::convert::Infallible;
+
+use abscissa_core::{Component, FrameworkError};
+use hyper::body::Incoming;
+use hyper::server::conn::http1;
+use hyper::service::service_fn;
+use hyper::{Request, Response, StatusCode};
+use serde::{Deserialize, Serialize};
+use tracing::{error, info};
+
+/// Abscissa component which runs a health endpoint.
+#[derive(Debug, Component)]
+#[cfg(feature = "health-endpoint")]
+pub struct HealthEndpoint {}
+
+#[cfg(feature = "health-endpoint")]
+impl HealthEndpoint {
+    /// Create the component.
+    pub fn new(config: &Config) -> Result<Self, FrameworkError> {
+        if let Some(addr) = config.endpoint_addr {
+            info!("Trying to open health endpoint at {}...", addr);
+
+            // Start the health endpoint server in a separate thread to avoid Tokio runtime issues
+            std::thread::spawn(move || {
+                let rt = tokio::runtime::Runtime::new().unwrap();
+                rt.block_on(async {
+                    if let Err(e) = Self::run_server(addr).await {
+                        error!("Health endpoint server failed: {}", e);
+                    }
+                });
+            });
+
+            info!("Opened health endpoint at {}", addr);
+        }
+
+        Ok(Self {})
+    }
+
+    async fn run_server(addr: SocketAddr) -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+        let listener = tokio::net::TcpListener::bind(addr).await?;
+        
+        loop {
+            let (stream, _) = listener.accept().await?;
+            let io = hyper_util::rt::TokioIo::new(stream);
+            
+            tokio::spawn(async move {
+                if let Err(err) = http1::Builder::new()
+                    .serve_connection(io, service_fn(Self::handle_request))
+                    .await
+                {
+                    error!("Failed to serve connection: {}", err);
+                }
+            });
+        }
+    }
+
+    async fn handle_request(req: Request<Incoming>) -> Result<Response<String>, Infallible> {
+        // Check if the request is for the health endpoint
+        if req.uri().path() != "/health" {
+            return Ok(Response::builder()
+                .status(StatusCode::NOT_FOUND)
+                .header("Content-Type", "application/json")
+                .body("{\"error\": \"Not Found\"}".to_string())
+                .unwrap());
```

Comment:
```suggestion
                .expect("response should build successfully"));
```
Consider replacing `unwrap()` with `expect(...)`